packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,173 @@
+
+const auditLogEventTypes = auditLogSchema.options.map(option => option.shape.eventType.value);
+
+export const AUDIT_LOG_CLICKHOUSE_OBJECT = z.object({
+  id: z.string(),
+  event_time: z.string(),
+  user_id: z.string(),
+  user_email: z.string(),
+  organization_id: z.string(),
+  event_action: z.enum(auditLogEventTypes as [string, ...string[]]),
+  metadata:
+    z.string().transform(x => auditLogSchema.parse(JSON.parse(x))) &&
+    z.string().transform(x => JSON.parse(x)),
+});
+
+export type AuditLogType = z.infer<typeof AUDIT_LOG_CLICKHOUSE_OBJECT>;
+
+const AuditLogClickhouseArrayModel = z.array(AUDIT_LOG_CLICKHOUSE_OBJECT);
+
+type AuditLogsArgs = {
+  selector: {
+    organization: string;
+  };
+  filter: {
+    userId: string;
+    startDate: Date;
+    endDate: Date;
+  };
+  pagination: {
+    limit: number;
+    offset: number;
+  };
+};
+
+type AuditLogRecordEvent = {
+  userId: string;
+  userEmail: string;
+  organizationId: string;
+  user: (User & { isAdmin: boolean }) | null;
+};
+
+@Injectable({
+  scope: Scope.Operation,
+  global: true,
+})
+export class AuditLogManager {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogManager' });
+  }
+
+  createLogAuditEvent(event: AuditLogEvent, record: AuditLogRecordEvent): void {
+    void this.internalCreateLogAuditEvent(event, record);
+  }
+
+  private async internalCreateLogAuditEvent(
+    event: AuditLogEvent,
+    record: AuditLogRecordEvent,
+  ): Promise<void> {
+    try {
+      const { eventType } = event;
+      const { organizationId, userEmail, userId } = record;
+      this.logger.debug('Creating a log audit event (event=%o)', event);
+
+      const parsedEvent = auditLogSchema.parse(event);
+      const metadata = {
+        user: record.user,
+        ...parsedEvent,
+      };
+
+      const eventMetadata = JSON.stringify(metadata);
+      const eventTime = new Date();
+
+      const values = [eventTime, userId, userEmail, organizationId, eventType, eventMetadata];
+
+      await this.clickHouse.insert({
+        query: sql`
+        INSERT INTO audit_log
+        (event_time, user_id, user_email, organization_id, event_action, metadata)
+        FORMAT CSV`,
+        data: [values],
+        timeout: 5000,
+        queryId: 'create-audit-log',
+      });
+    } catch (error) {
+      this.logger.error('Failed to create audit log event', error);
+      Sentry.captureException(error, {
+        extra: {
+          event,
+        },
+      });
+    }
+  }
+
+  async getPaginatedAuditLogs(
+    props: AuditLogsArgs,
+  ): Promise<{ total: number; data: AuditLogType[] }> {
+    this.logger.info(
+      'Getting paginated audit logs (organization=%s, filter=%o, pagination=%o)',
+      props.selector.organization,
+      props.filter,
+      props.pagination,
+    );
+
+    if (!props.selector.organization) {
```

Comment:
do we have a case where we are not going to pass this? 
I think you might not need this check, as PG will also error in case of empty/invalid uuid value